### DISCUSSION TITLE
nx-container for development?

### DISCUSSION THREAD

**Discussion by andefred** [gperdomor/oss#712] [Q&A]
Hi, this may be a stupid question, but how can I run nx-container for my development environment?

I have a client who really wants to keep using Docker containers to run the development environment (multiple nest microservices applications, and react frontend). They currently have it setup so that the volume is mounted and the app is run from there so that watching for changes works as normal.

However, my task is to move this to an NX repo for all the obvious reasons and I thought that it would be simple to just launch the same setup but with nx... I'm likely just missing something (I hope).

The ideal workflow in this case would be that the developer runs `docker-compose up` or similar (maybe using nx runner) and each application has it's Dockerfile specified in each app folder. How do I ensure that while developing, I don't have to rebuild the containers for code changes in either the applications or the libraries. It would be fine for package.json changes, but ideally not more than that.

Thanks!

**Comment by DaSchTour** [comment]
I have a somehow related question I think. I would like to have 2 configurations for my docker target. One for the CI to build and push to registry. That's working fine. The other should be for local testing and "load"? them into local docker as a container. I see that the are available with `docker images`.
